influxdb_store.go

```diff
@@ -647,8 +646,15 @@ func addChildren(root *Trace, children []*Trace) error {
 		if len(children) == 0 {
 			break
 		}
-		if try == retries {
-			return errMaxRetries
+		if try == retries { // At this point all children were added to their parent.
+
+			// Iterates over children(without parent found on `root`) and appends them as sub-traces to `root`.
+			for _, child := range children {
```

Comment:
Good call @slimsag, looks good to me :+1: —I've added it on beb73cdb509e607bf17919a0d8a990fe2730860a